### PR TITLE
Fix versioning in sample manifests

### PR DESCRIPTION
**Pull request by andrewsykim**
## Why are these changes needed?

Fix API versions specified in some sample manifests

## Checks

- [X] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ray-operator/config/samples/ray_v1alpha1_rayjob.shutdown.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: ray.io/v1
+apiVersion: ray.io/v1alpha1
```
Comment (andrewsykim) [inline ray-operator/config/samples/ray_v1alpha1_rayjob.shutdown.yaml:1]: @kevin85421 I wasn't sure if it was safe to delete the v1alpha1 manifests, so I kept it and just referenced v1alpha1 for now.

**Comment by andrewsykim** [comment]
I'm not sure why CI is failing:
```
Traceback (most recent call last):
  File "/workdir/tests/framework/prototype.py", line 654, in runtest
    self.cr_event.trigger()
  File "/workdir/tests/framework/prototype.py", line 187, in trigger
    self.exec()
  File "/workdir/tests/framework/prototype.py", line 206, in exec
    shell_subprocess_run(f"kubectl apply -n {self.namespace} -f {self.filepath}")
  File "/workdir/tests/framework/utils.py", line 368, in shell_subprocess_run
    return subprocess.run(command, shell=True, check=check).returncode
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'kubectl apply -n default -f /workdir/ray-operator/config/samples/ray_v1alpha1_rayjob.resources.yaml' returned non-zero exit status 1.
```

Seems potentially related to my change. This works locally for me though:
```
$ kubectl apply -f config/samples/ray_v1alpha1_rayjob.resources.yaml
rayjob.ray.io/rayjob-sample created
configmap/ray-job-code-sample created

```

**Comment by kevin85421** [comment]
I think there might be an issue with our CI infrastructure, and has no relationship with your PR. cc @architkulkarni, do you have the bandwidth to take a look since I will be on a business trip today? Thanks!

File: ray-operator/config/samples/ray_v1alpha1_rayjob.shutdown.yaml (hunk 1 above)
Comment (kevin85421) [reply ray-operator/config/samples/ray_v1alpha1_rayjob.shutdown.yaml:1]: You can directly rename the file, as we no longer require any YAML manifests with `apiVersion: ray.io/v1alpha1`. Moreover, our naming convention is `ray-{cluster | job | service}.{something_important_for_this_yaml}.yaml`. In this case, we can rename it to `ray-job.shutdown.yaml`.

Some YAMLs don't follow the naming style for their filenames. We can open another PR to improve that.

Comment (andrewsykim) [reply ray-operator/config/samples/ray_v1alpha1_rayjob.shutdown.yaml:1]: okay, thanks. Will update to follow this convention

**Comment by architkulkarni** [comment]
I've restarted the build in case it was some transient error.  

Just to record it, this was the error:

```
Image: "rayproject/ray:2.9.0" with ID "sha256:ee1c865e02572031f5c68be4184f21d929d5cd233289c83b2a10b9ce455eb75f" not yet present on node "kind-control-plane", loading...
ERROR: failed to load image: command "docker exec --privileged -i kind-control-plane ctr --namespace=k8s.io images import -" failed with error: exit status 1
Command Output: unpacking docker.io/rayproject/ray:2.9.0 (sha256:c0648f29a35efc1b2c3ee71393914fbab27e85ee4add32a1582d532159ae9f6d)...time="2024-01-22T16:47:05Z" level=info msg="apply failure, attempting cleanup" error="wrong diff id calculated on extraction \"sha256:db6680cb129d39fad6405b4fc83b612de4a228c95b437893af46d8b781ddd0f6\"" key="extract-983277564-yt-D sha256:3a03f09d212915b240e9d216069aba5652ed4765c7e4b098c65e71860d47b8e1"
ctr: wrong diff id calculated on extraction "sha256:db6680cb129d39fad6405b4fc83b612de4a228c95b437893af46d8b781ddd0f6"
```

**Comment by andrewsykim** [comment]
The `rayproject/ray:2.9.0 image showing different values for SHA256 is kind of weird. Is that locally built image or pull from internet? Is someone updating that?

File: ray-operator/config/samples/ray_v1alpha1_rayjob.shutdown.yaml (hunk 1 above)
Comment (andrewsykim) [reply ray-operator/config/samples/ray_v1alpha1_rayjob.shutdown.yaml]: updated to directly rename the files

**Comment by architkulkarni** [comment]
It's happening on this PR too https://github.com/ray-project/kuberay/pull/1858 so it's unrelated to your PR.

Moving the discussion of the CI issue to https://github.com/ray-project/kuberay/issues/1859

**Comment by kevin85421** [comment]
Would you mind rebasing with the master branch? Thanks!

**Comment by andrewsykim** [comment]
Done!

**Review by kevin85421 — COMMENTED** [review]
Please rename `ray-service.sample..yaml` to `ray-service.sample.yaml`.

**Comment by andrewsykim** [comment]
Thanks, fixed!

**Review by kevin85421 — APPROVED** [review]

**Pull request merged**